crates/proof-of-sql/src/proof_primitive/dory/dory_vmv_helper.rs

```diff
@@ -13,8 +21,50 @@ pub(super) fn compute_v_vec(a: &[F], L_vec: &[F], sigma: usize, nu: usize) -> Ve
         })
 }
 
+/// Converts a bls12-381 scalar to a u64 array.
+#[cfg(feature = "blitzar")]
+fn convert_scalar_to_padded_array(
+    scalars: &[ark_ff::Fp<MontBackend<ark_bls12_381::FrConfig, 4>, 4>],
+    num_columns: usize,
+    num_outputs: usize,
+) -> Vec<[u64; 4]> {
+    scalars
+        .iter()
+        .map(|&element| BigInt::<4>::from(element).0)
+        .chain(core::iter::repeat([0, 0, 0, 0]))
+        .take(num_columns * num_outputs)
+        .collect()
+}
+
 /// Compute the commitments to the rows of the matrix M that is derived from `a`.
 #[tracing::instrument(level = "debug", skip_all)]
+#[cfg(feature = "blitzar")]
+pub(super) fn compute_T_vec_prime(
+    a: &[F],
+    sigma: usize,
+    nu: usize,
+    prover_setup: &ProverSetup,
+) -> Vec<G1Affine> {
+    let num_columns = 1 << sigma;
+    let num_outputs = 1 << nu;
+    let data_size = std::mem::size_of::<F>();
+
+    let a_array = convert_scalar_to_padded_array(a, num_columns, num_outputs);
+    let a_transpose = transpose::transpose_for_fixed_msm(&a_array, 0, num_columns, data_size);
```

Comment:
Does this work?
```suggestion
    let a_transpose = transpose::transpose_for_fixed_msm(&a, 0, num_columns, data_size);
```

Reply:
No, passing `a` does not work because the trait bound `ark_ff::Fp<MontBackend<ark_bls12_381::FrConfig, 4>, 4>: AsBytes` is not satisfied.

The `transpose_for_fixed_msm` will pad to get to to the closest multiple of the `num_columns` above `offset + a.len()`. It does not know about the `num_outputs`. In this case we need pre-pad the array by the `num_outputs` in order to correctly do the transpose.

It does seem like the logic should be inside the `transpose_for_fixed_msm` function. Then it can just consider `rows` and `cols`. I can update the `transpose_for_fixed_msm` to do the padding an additional `num_outputs` parameter.